internal/controller/configmap.go

```diff
@@ -0,0 +1,163 @@
+}
+
+// getRBACScopes will return the Argo CD RBAC scopes.
+func getRBACScopes() string {
+	return common.ArgoCDDefaultRBACScopes
+}
+
+// getRBACDefaultPolicyCSV will return the Argo CD RBAC default policy CSV.
+func getDefaultRBACPolicy() string {
+	return common.ArgoCDDefaultRBACPolicy
+}
+
+// getRBACPolicyCSV will return the Argo CD RBAC policy CSV.
+func getRBACPolicyCSV(role *argoprojiov1alpha1.Role) string {
+	policy := ""
+
+	roleName := fmt.Sprintf("role:%s", role.ObjectMeta.Name)
+	policy += buildPolicyStringRules(role, roleName)
+
+	policy += buildPolicyStringSubjects(role, roleName)
+
+	return policy
+}
+
+// buildPolicyStringRules will build the policy string for Rules field of the given role.
+func buildPolicyStringRules(role *argoprojiov1alpha1.Role, roleName string) string {
+	policy := ""
+	for _, rule := range role.Spec.Rules {
+		resource := rule.Resource
+		for _, verb := range rule.Verbs {
+			for _, object := range rule.Objects {
+				policy += fmt.Sprintf("p, %s, %s, %s, %s, allow\n", roleName, resource, verb, object)
+			}
+		}
+	}
+	return policy
+}
+
+// buildPolicyStringSubjects will build the policy string for Subjects field of the given role.
+func buildPolicyStringSubjects(role *argoprojiov1alpha1.Role, roleName string) string {
+	policy := ""
+	for _, subject := range role.Spec.Subjects {
+		switch subject.Kind {
+		case "sso":
+			policy += fmt.Sprintf("g, %s, %s\n", subject.Name, roleName)
+		case "local":
+			policy += buildPolicyStringRules(role, subject.Name)
+		}
+	}
+	return policy
+}
+
+// newConfigMap will return a new ConfigMap resource.
+func newConfigMap() *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.ArgoCDRBACConfigMapName,
+			Namespace: common.ArgoCDRBACConfigMapNamespace,
+		},
+	}
+}
+
+// reconcileRBAC will ensure that the ArgoCD RBAC ConfigMap is present.
+func (r *RoleReconciler) reconcileRBAC(role *argoprojiov1alpha1.Role) error {
+	cm := newConfigMap()
+	if IsObjectFound(r.Client, cm.Namespace, cm.Name, cm) {
+		return r.reconcileRBACConfigMap(cm, role)
+	}
+	return r.createRBACConfigMap(cm, role)
+}
+
+// reconcileRBACConfigMap will ensure that the ArgoCD RBAC ConfigMap is up-to-date.
+func (r *RoleReconciler) reconcileRBACConfigMap(cm *corev1.ConfigMap, role *argoprojiov1alpha1.Role) error {
+	changed := false
+	overlayKey := fmt.Sprintf("policy.%s.csv", role.ObjectMeta.Name)
+
+	// Default Policy
+	if cm.Data[common.ArgoCDKeyRBACPolicyDefault] != getRBACPolicyDefault() {
+		cm.Data[common.ArgoCDKeyRBACPolicyDefault] = getRBACPolicyDefault()
+		changed = true
+	}
+	// Scopes
+	if cm.Data[common.ArgoCDKeyRBACScopes] != getRBACScopes() {
+		cm.Data[common.ArgoCDKeyRBACScopes] = getRBACScopes()
+		changed = true
+	}
+	// Default Policy String
+	if cm.Data[common.ArgoCDKeyRBACPolicyCSV] != getDefaultRBACPolicy() {
+		cm.Data[common.ArgoCDKeyRBACPolicyCSV] = getDefaultRBACPolicy()
+		changed = true
+	}
```

Comment:
this one i'm not sure. should we overwrite it in the operator? Same argument could be made on the user side, but at the same time it's less of a pain because I can just create another Role CR to put the content in

Reply:
The Operator should provide a single source of truth with the CRs so I would overwrite all manual changes to the `policy.csv` key